summary.Rmd

```diff
@@ -0,0 +1,40 @@
+Summary
+=======
+
+1. Use available tools
+    a. Use ready-made images (e.g. from Docker Hub)
+    b. Dockerfile generators (e.g. repo2docker)
+    c. Project templates (e.g. [cookie cutter](https://github.com/cookiecutter/cookiecutter))
+    d. Linters
+2. Use versioned images
+3. Format for clarity
+    a. Use indentation, line breaks and unabbreviated arguments
+    b. Clarity > brevity
+    c. Group related commands
+4. Document within the Dockerfile
```

Comment:
Same here: would you suggest to rename the subsections of this rule accordingly?

Or is consistency not an issue?

Reply:
I think so (although I will spell out "Clarity is better than brevity".

Reply:
Ok, so let's do another round on the subrules:

a. Add comments explaining the ~~steps~~ instructions [we don't use "step" anywhere for Dockerfile contents, AFAICS]
b. Use labels for metadata
c. Define versions, parameters, and paths once [New! What do you think is a better name?]
d. Include usage instructions [I prefer the more generic, because you might also add instructions for using a Makefile or running a custom build script]

I'll update the main document when we have a consensus here :-)

Reply:
Ok I've rephrased the existing sub-rules but do we really need "Define versions, parameters, and paths once"? This seems a bit like an unnecessary "do not contradict yourself" and I think for a summary we should try to limit ourselves to ideally 3 sub-points for the sake of brevity.

Reply:
I would take it out, because it's (for the most part) common sense.

Reply:
I'm OK with not having that subsection in the summary.